core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala

```diff
@@ -52,25 +53,40 @@ import org.apache.spark.sql.internal.SQLConf.SQLConfigBuilder
 import org.apache.spark.sql.policy.PolicyProperties
 import org.apache.spark.sql.sources._
 import org.apache.spark.sql.store.StoreUtils
-import org.apache.spark.sql.streaming.{LogicalDStreamPlan, WindowLogicalPlan}
+import org.apache.spark.sql.streaming.{LogicalDStreamPlan, StreamingQueryManager, WindowLogicalPlan}
 import org.apache.spark.sql.types._
 import org.apache.spark.streaming.Duration
 import org.apache.spark.unsafe.types.UTF8String
 import org.apache.spark.{Partition, SparkConf}
 
 
-class SnappySessionState(snappySession: SnappySession)
-    extends SessionState(snappySession) {
-
+class SnappySessionState(val snappySession: SnappySession)
+    extends SessionState(snappySession) with SnappyStrategies {
   self =>
 
   @transient
   val contextFunctions: SnappyContextFunctions = new SnappyContextFunctions
 
+  val sampleSnappyCase: PartialFunction[LogicalPlan, Seq[SparkPlan]] = {
+    case MarkerForCreateTableAsSelect(child) => PlanLater(child) :: Nil
+    case BypassRowLevelSecurity(child) => PlanLater(child) :: Nil
+    case _ => Nil
+  }
+
   protected lazy val snappySharedState: SnappySharedState = snappySession.sharedState
 
   private[internal] lazy val metadataHive = snappySharedState.metadataHive().newSession()
 
+  override lazy val streamingQueryManager: StreamingQueryManager = {
+    // Disabling `SnappyAggrerateStrategy` for streaming queries as it clashes with
+    // `StatefulAggregationStrategy` which is applied by spark for streaming queries. This
+    // implies that Snappydata aggregation optimisation will be turned off for any usage of
+    // this session including non-streaming queries.
+
+    HashAggregateSize.set(snappySession.sessionState.conf, "-1")
+    new StreamingQueryManager(snappySession)
```

Comment:
Will be better to call super.streamingQueryManager

Reply:
I tried "super.streamingQueryManager" earlier. However, it fails at compile time stating "super may not be used on lazy value streamingQueryManager". Based on my brief analysis it is possibly due to https://issues.scala-lang.org/browse/SI-1938.